cupy/core/_routines_manipulation.pyx

```diff
@@ -326,19 +325,19 @@ cpdef ndarray rollaxis(ndarray a, Py_ssize_t axis, Py_ssize_t start=0):
     return _transpose(a, axes)
 
 
-cpdef ndarray _reshape(ndarray self, vector.vector[Py_ssize_t] shape):
-    cdef vector.vector[Py_ssize_t] strides
+cpdef ndarray _reshape(ndarray self,
+                       const vector.vector[Py_ssize_t] &orig_shape):
+    cdef vector.vector[Py_ssize_t] shape, strides
     cdef ndarray newarray
-    shape = internal.infer_unknown_dimension(shape, self.size)
+    shape = internal.infer_unknown_dimension(orig_shape, self.size)
```

Comment:
The name `orig_shape` misled me.  Could you add comment to explain `orig_shape` is a shape that may contain -1?  Otherwise the variables could just be renamed:
```suggestion
    shape = internal.infer_unknown_dimension(shape_spec, self.size)
```
```suggestion
    inferred_shape = internal.infer_unknown_dimension(shape, self.size)
```

Reply:
thnaks, I will fix.